File: test-kotlin/helloworld-kotlinc/src/demo/Person.kt
package demo

import java.util.concurrent.locks.ReentrantReadWriteLock
import kotlin.concurrent.*

interface Helloable {
    fun sayHello(): Int
}

class Person (val name: String, var age: Int = 10, val lock: ReentrantReadWriteLock = ReentrantReadWriteLock()) : Helloable {
    public override fun sayHello(): Int {
        lock.write {
            println("${Thread.currentThread()}: Hello, ${name}:${age} with ${lock}")
            age += 1
            return age
        }
    }
}

File: test-kotlin/helloworld-scala/src/main/scala/person.scala
package demo

import java.util.concurrent.locks.ReentrantReadWriteLock
import demo.concurrent._

trait Helloable {
  def sayHello(): Integer
}

class Person(val name: String, var age: Int = 10, val lock: ReentrantReadWriteLock = new ReentrantReadWriteLock) extends Helloable {
  override def sayHello(): Integer =
    lock.write {
      println(s"${Thread.currentThread()}: Hello, ${name}:${age} with ${lock}")
      age += 1
      age
    }
}
